File: wallendar-app/app/src/main/java/com/td/wallendar/repositories/mappers/EventMapper.java
package com.td.wallendar.repositories.mappers;

import com.td.wallendar.dtos.response.ChargeResponse;
import com.td.wallendar.dtos.response.EventResponse;
import com.td.wallendar.dtos.response.EventsResponse;
import com.td.wallendar.models.Charge;
import com.td.wallendar.models.EqualCharge;
import com.td.wallendar.models.Event;
import com.td.wallendar.models.User;

import java.util.ArrayList;
import java.util.List;

public class EventMapper {

    public static List<Event> toModel(final EventsResponse eventsResponse) {
        final List<Event> events = new ArrayList<>();
        for (EventResponse eventResponse : eventsResponse.getEvents()) {
            events.add(toModel(eventResponse));
        }
        return events;
    }

    public static Event toModel(final EventResponse eventResponse) {
        List<User> debtors = new ArrayList<>();
        for (String debtor : eventResponse.getUsers()) {
            debtors.add(new User(debtor, "", "", ""));
        }

        List<Charge> charges = new ArrayList<>();
        for (ChargeResponse chargeResponse : eventResponse.getCharges()) {
            charges.add(new EqualCharge(
                    chargeResponse.getChargeId(),
                    new User(chargeResponse.getOwner(), "", "", ""),
                    chargeResponse.getTitle(),
                    new ArrayList<>(),
                    chargeResponse.getAmount(),
                    chargeResponse.getDate()
            ));
        }

        return new Event(
                1L,
                eventResponse.getTitle(),
                new User(eventResponse.getOwner(), "", "", ""),
                debtors,
                charges,
                new ArrayList<>(),
                eventResponse.getDate(),
                new ArrayList<>()
        );
    }
}

File: wallendar-app/app/src/main/java/com/td/wallendar/utils/mappers/EventMapper.kt
package com.td.wallendar.utils.mappers

import com.td.wallendar.dtos.response.EventResponse
import com.td.wallendar.models.Event
import java.util.*

object EventMapper {
    fun toModel(eventsResponse: MutableList<EventResponse>): MutableList<Event> {
        val groups: MutableList<Event> = ArrayList()
        for (group in eventsResponse) {
            groups.add(toModel(group))
        }
        return groups
    }

    fun toModel(eventResponse: EventResponse): Event {
        return Event(
                eventResponse.id,
                eventResponse.title,
                ApplicationUserMapper.toModel(eventResponse.owner),
                ApplicationUserMapper.toModel(eventResponse.members),
                ChargeMapper.toModel(eventResponse.charges),
                DebtMapper.toModel(eventResponse.debts),
                PaymentMapper.toModel(eventResponse.payments),
                eventResponse.date
        )
    }
}